media-proxy/src/mesh/conn_rdma_rx.cc

```diff
@@ -0,0 +1,154 @@
+#include "conn_rdma_rx.h"
+#include <stdexcept>
+#include <queue>
+
+namespace mesh::connection {
+
+RdmaRx::RdmaRx() : Rdma() {
+    _kind = Kind::receiver; // Set the Kind in the constructor
+    dir = direction::RX;    // Set the direction in the constructor
+}
+
+RdmaRx::~RdmaRx()
+{
+    // Any specific cleanup for Rx can go here
+}
+
+Result RdmaRx::configure(context::Context& ctx, const mcm_conn_param& request,
+                         libfabric_ctx *& dev_handle)
+{
+    return Rdma::configure(ctx, request, dev_handle);
+}
+
+Result RdmaRx::start_threads(context::Context& ctx)
+{
+
+    process_buffers_thread_ctx = context::WithCancel(ctx);
+    rdma_cq_thread_ctx = context::WithCancel(ctx);
+
+    try {
+        handle_process_buffers_thread = std::jthread(
+            [this]() { this->process_buffers_thread(this->process_buffers_thread_ctx); });
+    } catch (const std::system_error& e) {
+        log::fatal("Failed to start thread")("error", e.what());
+        return Result::error_thread_creation_failed;
+    }
+
+    try {
+        handle_rdma_cq_thread =
+            std::jthread([this]() { this->rdma_cq_thread(this->rdma_cq_thread_ctx); });
+    } catch (const std::system_error& e) {
+        log::fatal("Failed to start thread")("error", e.what());
```

Comment:
The use of `log::fatal` here confuses me because I understand that it exists the process right away.

Reply:
Can you replace `log::fatal` with `log::error` everywhere?

Reply:
fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/286/commits/794708adf96f927afbd18cdc93def4203541ba83